diff --git a/query_templates/netezza.tpl b/query_templates/netezza.tpl
index 75488d2..0ff3ce1 100755
--- a/query_templates/netezza.tpl
+++ b/query_templates/netezza.tpl
@@ -35,3 +35,5 @@
 define __LIMITA = "";
 define __LIMITB = "";
 define __LIMITC = "limit %d";
+define _BEGIN = "-- start query " + [_QUERY] + " in stream " + [_STREAM] + " using template " + [_TEMPLATE];
+define _END = "-- end query " + [_QUERY] + " in stream " + [_STREAM] + " using template " + [_TEMPLATE];
